### PR TITLE
Indicate staff made reservations

### PR DESCRIPTION
**Pull request by Wnt**
## Before this change
It was not possible to distinguish between staff made attendance reservations and citizen made reservations.

## After this change
Employee week view shows an asterisk and tooltip info on records that are made by staff.
image

This is achieved by adding a new boolean flag `staffCreated` into the `Reservation` response of `GET /attendance-reservations`. The new `ReservationResponse` datatype is also shared by other endpoints such as 
`GET /attendance-reservations/confirmed-days/daily`
`GET /attendances/units/${unitId}/children`

### CODE REVIEW

File: service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt [hunk 1]
```diff
@@ -216,22 +226,31 @@ fun Database.Read.getReservationDatesForChildrenInRange(
...
-    data class ReservationWithDate(val date: LocalDate, val reservation: Reservation)
+): Map<LocalDate, List<ReservationResponse>> {
+    data class ReservationWithDate(val date: LocalDate, val reservation: ReservationResponse)
     return createQuery(
             """
-        SELECT date, start_time, end_time
-        FROM attendance_reservation
-        WHERE between_start_and_end(:range, date)
-        AND child_id = :childId
+                SELECT ar.date,
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt:233]: weird indentation on first line

File: service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt [hunk 2]
```diff
@@ -540,17 +563,20 @@ SELECT pcd.child_id,
...
         (SELECT coalesce(jsonb_agg(json_build_object(
                'start', s.start_time,
-               'end', s.end_time)), '[]'::jsonb)
-        FROM (select ar.start_time, ar.end_time
-              FROM attendance_reservation ar
+               'end', s.end_time,
+               'staffCreated', s.staff_created)), '[]'::jsonb)
+        FROM (select ar.start_time, ar.end_time, eu.type = 'EMPLOYEE' as staff_created
+                FROM attendance_reservation ar
+                LEFT JOIN evaka_user eu ON ar.created_by = eu.id
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt:570]: created_by has not null so LEFT is unnecessary

File: service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt [hunk 3]
```diff
@@ -540,17 +563,20 @@ SELECT pcd.child_id,
...
               WHERE ar.child_id = pcd.child_id
                 AND ar.date = :examinationDate) s)
            AS reservations,
        -- absence roll up
        (SELECT coalesce(jsonb_agg(json_build_object(
                'category', s.category)), '[]'::jsonb)
-        FROM (SELECT ab.category
+        FROM (SELECT ab.category, eu.type = 'EMPLOYEE' as staff_created
               FROM absence ab
+              LEFT JOIN evaka_user eu ON ab.modified_by = eu.id
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt:579]: modified_by has not null so LEFT is unnecessary

File: service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt [hunk 4]
```diff
@@ -894,9 +899,10 @@ SELECT
         SELECT jsonb_agg(jsonb_build_object(
             'date', ar.date,
             'startTime', ar.start_time,
-            'endTime', ar.end_time
+            'endTime', ar.end_time,
+            'staffCreated', eu.type = 'EMPLOYEE'
         ) ORDER BY ar.date, ar.start_time)
-        FROM attendance_reservation ar WHERE ar.child_id = p.id AND between_start_and_end(:dateRange, ar.date)
+        FROM attendance_reservation ar LEFT JOIN evaka_user eu ON ar.created_by = eu.id WHERE ar.child_id = p.id AND between_start_and_end(:dateRange, ar.date)
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt:905]: created_by has not null so LEFT is unnecessary. would also add line breaks before join and before where

File: service/src/main/kotlin/fi/espoo/evaka/daycare/service/AbsenceQueries.kt [hunk 5]
```diff
@@ -507,7 +507,7 @@ AND daterange(bc.start_date, bc.end_date, '[]') && :period
         .groupBy({ it.first }, { it.second })
 
 data class ChildReservation(
-    val reservation: Reservation,
+    val reservation: ReservationResponse,
```
Comment (Joosakur) [inline service/src/main/kotlin/fi/espoo/evaka/daycare/service/AbsenceQueries.kt:510]: here we already have `val createdByEvakaUserType: EvakaUserType` so we probably don't need both that and the new boolean. could one of them be removed?

File: frontend/src/employee-mobile-frontend/child-attendance/ChildSubListItem.tsx [hunk 6]
```diff
@@ -115,7 +115,7 @@ export default React.memo(function ChildSubListItem({
     if (reservationData.outOnBackupPlacement)
       return [i18n.attendances.confirmedDays.inOtherUnit]
 
-    const [withTimes, noTimes] = partition(
+    const [withTimes] = partition(
```
Comment (Joosakur) [inline frontend/src/employee-mobile-frontend/child-attendance/ChildSubListItem.tsx:118]: if we don't need the other half then `filter` makes more sense

File: frontend/src/e2e-test/specs/6_mobile/confirmed-day-reservations.spec.ts [hunk 7]
```diff
@@ -393,6 +394,17 @@ async function insertConfirmedDaysTestData() {
       end: LocalTime.of(16, 45)
     }
   }).save()
+
+  // await Fixture.attendanceReservation({
```
Comment (Joosakur) [inline frontend/src/e2e-test/specs/6_mobile/confirmed-day-reservations.spec.ts:398]: was this forgotten?

File: service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt (hunk 1 above)
Comment (Wnt) [reply service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt]: Fixed!

File: service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt (hunk 2 above)
Comment (Wnt) [reply service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt]: Fixed!

File: service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt (hunk 4 above)
Comment (Wnt) [reply service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt]: Fixed!

File: frontend/src/employee-mobile-frontend/child-attendance/ChildSubListItem.tsx (hunk 6 above)
Comment (Wnt) [reply frontend/src/employee-mobile-frontend/child-attendance/ChildSubListItem.tsx]: Changed to `filter()`!

File: frontend/src/e2e-test/specs/6_mobile/confirmed-day-reservations.spec.ts (hunk 7 above)
Comment (Wnt) [reply frontend/src/e2e-test/specs/6_mobile/confirmed-day-reservations.spec.ts]: Yes, reverted these leftover draft changes!

File: service/src/main/kotlin/fi/espoo/evaka/daycare/service/AbsenceQueries.kt (hunk 5 above)
Comment (Wnt) [reply service/src/main/kotlin/fi/espoo/evaka/daycare/service/AbsenceQueries.kt]: Indeed! I changed this back to using `Reservation` data type + bought back the front-end `Reservation` deserializer code etc.

**Review by Joosakur — APPROVED** [review]

**Pull request merged**